### PR TITLE
Adding stellar mass-gas metallicity relation from Chartab+21

### PR DESCRIPTION
**Pull request by correac**

### CODE REVIEW

**Comment by correac** [comment]
Hi! I'm adding some data for the stellar mass- gas metallicity (log(O/H)+12) at redshift 1.5 and 2. Only including the bestfit lines done by Chartab+21 to some galaxy stacks. It would be useful to have it included in the comparison-data folder.

**Comment by JBorrow** [comment]
Apologies for multi-z data we now use the `MultiRedshiftObservationalData` container, see https://velociraptor-python.readthedocs.io/en/latest/observational_data/creating_data.html#multi-redshift-data. If you could change this PR to use that API it would be helpful (see e.g. the multi-z stellar mass functions that we changed recently for an example)

**Comment by correac** [comment]
Oh no problem! I had an initial version with multiredshift but then I removed it because I thought you were not using it.

**Comment by correac** [comment]
Done. Comments welcome! :)

File: data/GalaxyStellarMassGasMetallicity/conversion/convertChartab2021.py [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+
+# Meta-data
+citation = "Chartab et al. (2021)"
+bibcode = "2021arXiv210101706C"
+name = "MOSDEF Survey: gas-phase metallicity of galaxies at 1.4 <= z <= 2.6"
+plot_as = "line"
+h = h_sim
+
+# Create, format, and save the data
+StellarMass_vs_GasMetallicity()
```
Comment (JBorrow) [inline data/GalaxyStellarMassGasMetallicity/conversion/convertChartab2021.py:107]: Not sure we need this to be a function, espsecially when it takes no parameters and uses global variables. Perhaps move the contents of `StelarMass_vs_GasMetallicity()` to the bottom here.

File: data/GalaxyStellarMassGasMetallicity/conversion/convertChartab2021.py [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+            Z_gas = (Z_asm + alpha * M_star) * unyt.dimensionless  # 12 + log(O/H)
+            M_star = 10 ** M_star * unyt.Solar_Mass
+
+        if z == 2 or z == 2.5:
+
+            Z_asm = 4.84
+            alpha = 0.35
+            M_star = np.arange(log10_M_star_min, log10_M_star_max, 0.2)
+            Z_gas = (Z_asm + alpha * M_star) * unyt.dimensionless  # 12 + log(O/H)
+            M_star = 10 ** M_star * unyt.Solar_Mass
```
Comment (JBorrow) [inline data/GalaxyStellarMassGasMetallicity/conversion/convertChartab2021.py:61]: This block can fall through. Can we make it:

```
if z < 2:
    Z_asm = 6.29
    alpha = 0.21
else:
    Z_asm = 4.84
    alpha = 0.35

log_M_star = np.arange(log10_M_star_min, log10_M_star_max, 0.2)
Z_gas = (Z_asm + alpha * log_M_star) * unyt.dimensionless
M_star = (10**log_M_star) * unyt.Solar_Mass
```

File: data/GalaxyStellarMassGasMetallicity/conversion/convertChartab2021.py [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+
+        processed.associate_x(
+            M_star, scatter=None, comoving=True, description="Galaxy Stellar Mass"
+        )
+        processed.associate_y(
+            Z_gas, scatter=None, comoving=True, description="Gas phase metallicity"
```
Comment (JBorrow) [inline data/GalaxyStellarMassGasMetallicity/conversion/convertChartab2021.py:67]: Can we include here infromation about the weird units?

**Review by JBorrow — COMMENTED** [review]
Looks good just a few minor comments.

File: data/GalaxyStellarMassGasMetallicity/conversion/convertChartab2021.py (hunk 3 above)
Comment (correac) [reply data/GalaxyStellarMassGasMetallicity/conversion/convertChartab2021.py]: Yes, I added some info in comments.

File: data/GalaxyStellarMassGasMetallicity/conversion/convertChartab2021.py (hunk 2 above)
Comment (correac) [reply data/GalaxyStellarMassGasMetallicity/conversion/convertChartab2021.py]: Sure!

File: data/GalaxyStellarMassGasMetallicity/conversion/convertChartab2021.py (hunk 1 above)
Comment (correac) [reply data/GalaxyStellarMassGasMetallicity/conversion/convertChartab2021.py]: Yes!

**Comment by JBorrow** [comment]
Excellent, thank you!

**Comment by correac** [comment]
Thank you!

**Pull request merged**